### PR TITLE
feat add support for different s3 providers

### PR DESCRIPTION
**Pull request by andrzejmikolajczakds**
### Description

Add other than amazon s3 providers as cloud storage

### Checklist

- [*] Tests have been added (if applicable, ie. when operator codes are added or modified)
- [*] Relevant docs have been added or modified (if applicable, ie. when new features are added or current features are modified)

Relevant issue #122

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -657,6 +658,60 @@ Here is an IAM policy to attach for Gatling Pod to interact with Amazon S3 bucke
 - Replace `BUCKET_NAME` above with your bucket name
 - To know more about the ways to supply rclone with a set of AWS credentials, please check [this](https://rclone.org/s3/#configuration).
 
+#### Set Different from Amazon S3 Provider as Cloud Storage
+
+This section provides guidance on setting up any cloud storage provider that supports the S3 API.
+In this example suppose you want to store Gatling reports to a bucket named `gatling-operator-reports` in OHV's S3 provider, specifically in the `de` region.
```
Comment (gold-kou) [inline docs/user-guide.md:664]:
```suggestion
In this example suppose you want to store Gatling reports to a bucket named `gatling-operator-reports` in OVH's S3 provider, specifically in the `de` region. 
```

File: pkg/cloudstorages/cloudstorage.go [hunk 2]
```diff
@@ -27,7 +27,9 @@ func GetProvider(provider string, args ...EnvVars) *CloudStorageProvider {
 		var csp CloudStorageProvider
 		switch provider {
 		case "aws":
-			csp = &AWSCloudStorageProvider{providerName: provider}
+			csp = &S3CloudStorageProvider{providerName: provider}
```
Comment (gold-kou) [inline pkg/cloudstorages/cloudstorage.go:30]: The backward compatibility for `aws` looks maintained.
Sounds good.

File: docs/user-guide.md [hunk 3]
```diff
@@ -657,6 +658,60 @@ Here is an IAM policy to attach for Gatling Pod to interact with Amazon S3 bucke
 - Replace `BUCKET_NAME` above with your bucket name
 - To know more about the ways to supply rclone with a set of AWS credentials, please check [this](https://rclone.org/s3/#configuration).
 
+#### Set Different from Amazon S3 Provider as Cloud Storage
```
Comment (gold-kou) [inline docs/user-guide.md:661]: This sentence might be a bit weird.
How about just `Set S3 as Cloud Storage` ?

**Review by gold-kou — COMMENTED** [review]
Thanks for your contrib!
Just nits.

**Review by gold-kou — APPROVED** [review]
LGTM

**Review by kane8n — APPROVED** [review]
lgtm!

**Pull request merged**